File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/historical_oi.kt
// To parse the JSON, install Klaxon and do:
//
//   val historicalOi = HistoricalOi.fromJson(jsonString)

package quicktype

import com.beust.klaxon.*

private val klaxon = Klaxon()

data class HistoricalOi (
    @Json(name = "\$ref")
    val ref: String,

    @Json(name = "\$schema")
    val schema: String,

    val definitions: Definitions
) {
    public fun toJson() = klaxon.toJsonString(this)

    companion object {
        public fun fromJson(json: String) = klaxon.parse<HistoricalOi>(json)
    }
}

data class Definitions (
    @Json(name = "Candle")
    val candle: Candle,

    @Json(name = "Data")
    val data: Data,

    @Json(name = "HistoricalOi")
    val historicalOi: HistoricalOiClass
)

data class Candle (
    val anyOf: List<AnyOf>,
    val title: String
)

data class AnyOf (
    val type: String
)

data class Data (
    val additionalProperties: Boolean,
    val properties: DataProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class DataProperties (
    val candles: Candles
)

data class Candles (
    val items: Items,
    val type: String
)

data class Items (
    val items: DataClass,
    val type: String
)

data class DataClass (
    @Json(name = "\$ref")
    val ref: String
)

data class HistoricalOiClass (
    val additionalProperties: Boolean,
    val properties: HistoricalOiProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class HistoricalOiProperties (
    val data: DataClass,
    val status: AnyOf
)

File: JsonSchemaAndStructs/KiteConnect-Dart-Structs/historical_oi.dart
// To parse this JSON data, do
//
//     final historicalOi = historicalOiFromMap(jsonString);

import 'package:freezed_annotation/freezed_annotation.dart';
import 'dart:convert';

part 'historical_oi.freezed.dart';
part 'historical_oi.g.dart';

@freezed
abstract class HistoricalOi with _$HistoricalOi {
    const factory HistoricalOi({
        Data data,
        String status,
    }) = _HistoricalOi;

    factory HistoricalOi.fromJson(Map<String, dynamic> json) => _$HistoricalOiFromJson(json);
}

@freezed
abstract class Data with _$Data {
    const factory Data({
        List<List<dynamic>> candles,
    }) = _Data;

    factory Data.fromJson(Map<String, dynamic> json) => _$DataFromJson(json);
}
